File: server/src/main/java/com/edutech/progressive/entity/Cricketer.java
package com.edutech.progressive.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "cricketer")
public class Cricketer implements Comparable<Cricketer> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cricketer_id")
    private int cricketerId;

    @ManyToOne
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "cricketer_name")
    private String cricketerName;

    @Column(name = "age")
    private int age;

    private String nationality;
    private int experience;
    private String role;

    @Column(name = "total_runs")
    private int totalRuns;

    @Column(name = "total_wickets")
    private int totalWickets;

    public Cricketer() {
    }

    public Cricketer(int cricketerId, Team team, String cricketerName, int age, String nationality, int experience,
            String role, int totalRuns, int totalWickets) {
        this.cricketerId = cricketerId;
        this.team = team;
        this.cricketerName = cricketerName;
        this.age = age;
        this.nationality = nationality;
        this.experience = experience;
        this.role = role;
        this.totalRuns = totalRuns;
        this.totalWickets = totalWickets;
    }

    public int getCricketerId() {
        return cricketerId;
    }

    public void setCricketerId(int cricketerId) {
        this.cricketerId = cricketerId;
    }

    public Team getTeam() {
        return team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }

    public String getCricketerName() {
        return cricketerName;
    }

    public void setCricketerName(String cricketerName) {
        this.cricketerName = cricketerName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getNationality() {
        return nationality;
    }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public int getExperience() {
        return experience;
    }

    public void setExperience(int experience) {
        this.experience = experience;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public int getTotalRuns() {
        return totalRuns;
    }

    public void setTotalRuns(int totalRuns) {
        this.totalRuns = totalRuns;
    }

    public int getTotalWickets() {
        return totalWickets;
    }

    public void setTotalWickets(int totalWickets) {
        this.totalWickets = totalWickets;
    }

    @Override
    public int compareTo(Cricketer o) {
        return Integer.compare(this.getExperience(), o.getExperience());
    }
}

File: client/src/app/ipl/types/Cricketer.ts
import { Team } from "./Team";

export class Cricketer {
  cricketerId: number;
  teamId: number;
  team?: Team;
  cricketerName: string;
  age: number;
  nationality: string;
  experience: number;
  role: string;
  totalRuns: number;
  totalWickets: number;

  constructor(
    cricketerId: number,
    teamId: number,
    cricketerName: string,
    age: number,
    nationality: string,
    experience: number,
    role: string,
    totalRuns: number,
    totalWickets: number
  );

  constructor(
    cricketerId: number,
    cricketerName: string,
    age: number,
    nationality: string,
    experience: number,
    role: string,
    totalRuns: number,
    totalWickets: number,
    team: Team
  );

  constructor(
    cricketerId: number,
    secondParam: number | string,
    thirdParam: string | number,
    fourthParam: number | string,
    fifthParam: string | number,
    sixthParam: number | string,
    seventhParam: string | number,
    eighthParam: number,
    ninthParam: number | Team,
    tenthParam?: number
  ) {
    this.cricketerId = cricketerId;

    if (typeof secondParam === "number") {
      this.teamId = secondParam;
      this.cricketerName = thirdParam as string;
      this.age = fourthParam as number;
      this.nationality = fifthParam as string;
      this.experience = sixthParam as number;
      this.role = seventhParam as string;
      this.totalRuns = eighthParam;
      this.totalWickets = tenthParam as number;
    } else {
      this.cricketerName = secondParam;
      this.age = thirdParam as number;
      this.nationality = fourthParam as string;
      this.experience = fifthParam as number;
      this.role = sixthParam as string;
      this.totalRuns = seventhParam as number;
      this.totalWickets = eighthParam;
      this.team = ninthParam as Team;
      this.teamId = this.team.teamId;
    }
  }

  displayInfo(): void {
    console.log(`Cricketer ID: ${this.cricketerId}`);
    console.log(`Team ID: ${this.teamId}`);
    console.log(`Cricketer Name: ${this.cricketerName}`);
  }
}